### ISSUE TITLE
Hook `useRouter` returns router without `dismiss`, etc. methods

### ISSUE THREAD

**Issue by DanielChomat** [expo/expo#27273]
### Minimal reproducible example

https://github.com/DanielChomat/expo-router-MRE-dismiss

### Which package manager are you using? (Yarn is recommended)

yarn

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

[Documentation](https://docs.expo.dev/router/advanced/stack/#dismiss-action) states that there are additional methods  in `router` like `dismiss`, `dismissAll` and `canDismiss`. I also see them in the expo-router on [GitHub](https://github.com/expo/expo/blob/main/packages/expo-router/build/types.d.ts#L14).

However I do not see them in my installed version of expo-router, does anyone know why is that? 
These methods would be very helpful in my usecase.

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.10.0 - /usr/local/bin/node
      Yarn: 1.22.21 - /usr/local/bin/yarn
      npm: 10.2.3 - /usr/local/bin/npm
      Watchman: 2023.11.27.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.7 => 50.0.7 
      expo-router: ~3.4.7 => 3.4.7 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
      react-native-web: ~0.19.6 => 0.19.10 
    Expo Workflow: managed

```

**Comment by tomsenior** [comment]
I've got this issue too. Could really use the dismiss functions, not sure if i'm missing something it looks like it should be available from the docs

**Comment by utkarsh1097** [comment]
It seems like `dismiss`, `canDismiss`, and `dismissAll` have not been released yet. Reference: [commit diff](https://github.com/expo/expo/commit/bf2da2bdb9388d5e20a1a3df02f70f9cce4654ad#diff-860a33690cd969a01bdf910036342b764b2d99b11bc62ac7c14cf8dc9fc90a39)

Considering that `expo-router`'s [latest version is still 3.4.7](https://github.com/expo/expo/blob/sdk-50/packages/expo-router/package.json#L3) and that it released a couple of weeks ago, the new `Router` API changes will likely reflect in a new version release.

**Comment by DanielChomat** [comment]
> It seems that `dismiss`, `canDismiss`, and `dismissAll` have not been released yet. Reference: [commit diff](https://github.com/expo/expo/commit/bf2da2bdb9388d5e20a1a3df02f70f9cce4654ad#diff-860a33690cd969a01bdf910036342b764b2d99b11bc62ac7c14cf8dc9fc90a39)
> 
> 
> 
> Considering that `expo-router`'s [latest version is still 3.4.7](https://github.com/expo/expo/blob/sdk-50/packages/expo-router/package.json#L3) and that it released a couple of weeks ago, the new `Router` API changes will likely reflect in a new version release.

Thanks @utkarsh1097 for the clarification, figured since it's already in the documentation, it would be merged already.

Will wait until the next release then.

Thanks @marklawlor.

**Comment by utkarsh1097** [comment]
Yes, even I originally read about it in the docs but couldn't use the methods, tried looking it up here, and ended up finding this issue before doing a deep dive :)

**Comment by marklawlor** [comment]
Fixed in `3.4.8`. Sorry that about, docs were accidentally published before the package.
